pkg/server/server.go

```diff
@@ -673,6 +673,27 @@ func (s *Server) indexHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// This Content Security Policy (CSP) applies to Console web application resources.
+	// Console CSP is deployed in report-only mode via "Content-Security-Policy-Report-Only" header.
+	// See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for details on CSP specification.
+	cspSources := "'self'"
+	if s.K8sMode == "off-cluster" {
+		// Console local development involves a webpack server running on port 8080
+		cspSources = cspSources + " http://localhost:8080 ws://localhost:8080"
+	}
+	cspDirectives := []string{
+		fmt.Sprintf("default-src %s", cspSources),
+		fmt.Sprintf("base-uri %s", cspSources),
+		fmt.Sprintf("img-src %s data:", cspSources),
+		fmt.Sprintf("font-src %s data:", cspSources),
+		fmt.Sprintf("script-src %s 'unsafe-eval'", cspSources),
+		fmt.Sprintf("style-src %s 'unsafe-inline'", cspSources),
+		"frame-src 'none'",
+		"frame-ancestors 'none'",
+		"object-src 'none'",
+	}
```

Comment:
Great start.

I could see some of these directives breaking plugins. As we've talked about, we'll minimally need a mechanism for plugins to add values to `cspSources` and a plan to transition to enforcing.

Reply:
@spadgett 
> As we've talked about, we'll minimally need a mechanism for plugins to add values to cspSources

Sounds like an API change for the ConsolePlugin CRD

> plan to transition to enforcing.

I feel like by the end of 4.18 we should have a clear picture on:
1. What the CSP which we want to enforce should look like
2. API change for the ConsolePlugin